### PR TITLE
Allow reuse of shared actuators depending on BrokerClient

### LINKED ISSUE #9996: Actuators are cumbersome to maintain and extend
**Description**

We have multiple open issues to extend or improve the actuators:
- https://github.com/camunda/zeebe/issues/9961
- https://github.com/camunda/zeebe/issues/9630
- https://github.com/camunda/zeebe/issues/9633

Implementing actuators can be very cumbersome due to the need for `SpringGatewayBridge` and `SpringBrokerBridge` which leads to code duplication (see `RebalancingService`).

There are other issues, for example the /rebalance actuator only works on a standalone gateway and on brokers with an embedded gateway. There's no technical reason it shouldn't work on standalone brokers without the embedded gateway, the only reason it doesn't work currently is that no one builds a `BrokerClient`.

I think we should spend some time to improve this. I've tried out a couple of approaches but they require a decent amount of refactoring.

My approach would be this: 

- [ ] Let the `dist` module be in charge of creating `Broker` , `Gateway` , `BrokerClient` , `AtomixCluster` and `ActorScheduler`. For `StandaloneGateway` we are doing most of that already but not for `StandaloneBroker`. 
- [ ] Actuators in `dist` have direct access to a `BrokerClient` because it can be injected by Spring, which removes the need for the spring bridges and code duplication.

related to https://github.com/camunda/zeebe/issues/8845

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR provides the last remaining bit to #9996 by having each application provide a `BrokerClient` bean, allowing us to create shared actuators which do not need any bridge. To showcase, the `RebalancingService` was collapsed into a single class.

## Related issues

closes #9996 
depends on #10354 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
Forgot to start the broker client on the actor scheduler is started. This is kind of hacky, as the `StandaloneBroker` also needs the client to be injected so it can start it after the actor scheduler.

File: qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/RebalanceActuator.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ * <p>Adding a new method is simple: simply define the input/output here as you normally would, and
+ * make sure to add the correct JSON encoding headers (`Accept` for the response type,
+ * `Content-Type` if there's a body to send). See {@link LoggersActuator} for a more complete
+ * example.
+ */
+public interface RebalanceActuator {
+
+  /**
+   * Returns a {@link RebalanceActuator} instance using the given node as upstream. This only
+   * accepts {@link ZeebeBrokerNode} at the moment, as only the broker has this actuator. It can be
```
Comment (deepthidevaki) [inline qa/util/src/main/java/io/camunda/zeebe/qa/util/actuator/RebalanceActuator.java:36]: :x: Did you mean "only accepts ZeebeGatewayNode" ?

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/RebalancingEndpointIT.java [hunk 2]
```diff
@@ -64,17 +55,10 @@ void shouldRebalanceCluster() throws IOException, InterruptedException {
...
-    final var request =
-        HttpRequest.newBuilder()
-            .POST(BodyPublishers.noBody())
-            .uri(URI.create("http://" + monitoringAddress + "/actuator/rebalance"))
-            .build();
-    final var response = httpClient.send(request, BodyHandlers.discarding());
-    assertThat(response.statusCode()).isEqualTo(200);
+  private void triggerRebalancing() {
+    final var gateway = (ZeebeContainer) cluster.getBrokers().get(0);
+    //    final ZeebeGatewayNode<?> gateway = cluster.getAvailableGateway();
```
Comment (deepthidevaki) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/RebalancingEndpointIT.java:60]: :x:  Here, we can use `gateway = cluster.getAvailableGateway()`, right?

**Review by deepthidevaki — APPROVED** [review]
:100: Looks great. 
Just two minor comments. Please take a look before merging.

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**